tests/aircall/test_aircall.py

```diff
@@ -247,3 +302,69 @@ def test_datasource():
         limit=1,
     )
     assert ds.dataset == 'calls'
+
+
+def test_get_status_no_secrets(con, remove_secrets):
+    """
+    It should fail if no secrets are provided
+    """
+    assert con.get_status().status is False
+
+
+def test_get_status_secrets_error(mocker, con):
+    """
+    It should fail if secrets can't be retrieved
+    """
+    mocker.patch(f'{import_path}.OAuth2Connector.get_access_token', side_effect=Exception)
+    assert con.get_status().status is False
+
+
+def test_get_status_api_down(mocker, con):
+    """
+    It should fail if the third-party api is down.
+    """
+    mocker.patch.object(AircallConnector, 'get_access_token', side_effect=HttpError)
+    assert con.get_status().status is False
```

Comment:
Thanks for writing these tests :+1: 
We can see here that they are quite similar than the one written in the GoogleSheets connector.
It's OK for now, but let's keep that in mind so if we have to write these again, we'll try to factor it into OAuth2Connector

Reply:
Totally agree, as we'll add other oauth connectors. I created a card in the Backlog Tech in our Board.